RequestService/RequestService.Handlers/PostRequestForHelpHandler.cs

```diff
@@ -0,0 +1,369 @@
+                                };
+                        await _communicationService.RequestCommunication(new RequestCommunicationRequest()
+                        {
+                            CommunicationJob = new CommunicationJob { CommunicationJobType = CommunicationJobTypes.RequestorTaskConfirmation },
+                            JobID = null,
+                            AdditionalParameters = additionalParameters,
+                            RequestID = requestId
+                        }, cancellationToken);
+                        break;
+                    case NewTaskAction.SetStatusToOpen:
+                        await _repository.UpdateAllJobStatusToOpenForRequestAsync(requestId, -1, cancellationToken);
+                        break;
+                }
+
+            }
+        }
+
+        private async Task<int> ProcessRequest(HelpRequestDetail helpRequestDetail, 
+            GetRequestHelpFormVariantResponse formVariant, 
+            Fulfillable fulfillable,
+            bool suppressRecipientPersonalDetails, 
+            bool multiVolunteers,
+            bool repeat, CancellationToken cancellationToken)
+        {
+            var actions = _groupService.GetNewRequestActions(new GetNewRequestActionsRequest()
+            {
+                HelpRequest = helpRequestDetail.HelpRequest,
+                NewJobsRequest = helpRequestDetail.NewJobsRequest
+            }, cancellationToken).Result;
+
+            if (actions == null)
+            {
+                throw new Exception("No new request actions returned");
+            }
+
+            int requestId = await _repository.NewHelpRequestAsync(new PostNewRequestForHelpRequest()
+            {
+                HelpRequest = helpRequestDetail.HelpRequest,
+                NewJobsRequest = helpRequestDetail.NewJobsRequest
+            }, fulfillable, formVariant.RequestorDefinedByGroup, suppressRecipientPersonalDetails, multiVolunteers, repeat);
+
+            if (requestId == 0)
+            {
+                throw new Exception("Error in saving request");
+            }
+
+            await ProcessRequestActions(actions, requestId, cancellationToken);
+
+            return requestId;
+        }
+
+        public async Task<PostRequestForHelpResponse> Handle(PostRequestForHelpRequest request, CancellationToken cancellationToken)
+        {
+            PostRequestForHelpResponse response = new PostRequestForHelpResponse()
+            {
+                RequestIDs = new List<int>()
+            };
+            var firstHelpRequestDetail = request.HelpRequestDetails.First();
+            bool invalidPostCode = await InvalidPostCode(firstHelpRequestDetail.HelpRequest, cancellationToken);
+
+            if(invalidPostCode)
+            {
+                return GetPostRequestForHelpResponse(new List<int> { -1 }, Fulfillable.Rejected_InvalidPostcode);
+            }
+
+            bool multiVolunteers = firstHelpRequestDetail.MultiVolunteer;
```

Comment:
Slightly confused about where these MultiVolunteer and Repeat flags are coming from and where they are going.  Feels like they should either be used to control entry to AddMultiAndRepeats, or returned from there.